nncf/experimental/post_training/api/metric.py

```diff
@@ -134,8 +134,13 @@ def update(self, outputs: NNCFData, targets: NNCFData) -> None:
         outputs, targets = self._extract(outputs, targets)
 
         if outputs.ndim != 2:
-            raise ValueError('The accuracy metric should be calculated on 2d outputs. '
-                             f'However, the outputs has ndim={outputs.ndim}.')
+            if outputs.ndim == 4 and (outputs.shape[2] == 1 and outputs.shape[3] == 1):
+                # densenet outputs (1,1000,1,1). It is able to change tensor shape.
+                outputs = outputs.squeeze(axis=(2, 3))
+            else:
+                raise ValueError('The accuracy metric should be calculated on 2d outputs. '
+                                 f'However, the outputs has ndim={outputs.ndim}.')
```

Comment:
Please avoid nested if-else statements for the better readability.

```
if:
    ~
elif:
    ~
else:
    ~
```

Reply:
Done